src/client-side-metrics/operation-metrics-collector.ts

```diff
@@ -281,6 +288,25 @@ export class OperationMetricsCollector {
     }
   }
 
+  /**
+   * Called when a row from the Bigtable stream reaches the application user.
+   *
+   * This method is used to calculate the latency experienced by the application
+   * when reading rows from a Bigtable stream. It records the time between the
+   * previous row being received and the current row reaching the user. These
+   * latencies are then collected and reported as `applicationBlockingLatencies`
+   * when the operation completes.
+   */
+  onRowReachesUser() {
+    const currentTime = new Date();
```

Comment:
Maybe you should check that you're in the expected state here?

Reply:
Alright. I added a guard here.